ppoprf/src/ppoprf.rs

```diff
@@ -87,6 +135,115 @@ impl ProofDLEQ {
+    challenge_transcript.extend_from_slice(z.compress().as_bytes()); //a1 = z
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a2)
+    challenge_transcript.extend_from_slice(t2.compress().as_bytes()); //a2 = t2
+    challenge_transcript.extend_from_slice(compressed_point_len_slice); //len(a3)
+    challenge_transcript.extend_from_slice(t3.compress().as_bytes()); //a3 = t3
+
+    //c = G.HashToScalar(challengeTranscript)
+    let c = ProofDLEQ::hash_to_scalar(&challenge_transcript, "Challenge");
+
+    //verified = (expectedC == c)
+    self.c == c
+  }
+
+  fn compute_composites(
+    key: Option<RistrettoScalar>,
+    b: &RistrettoPoint,
+    c: &[RistrettoPoint],
+    d: &[RistrettoPoint],
+  ) -> (RistrettoPoint, RistrettoPoint) {
+    if c.len() != d.len() {
+      panic!("C and D have a different number of elements!");
+    }
+
+    // We use the Partially-punctureable Oblivious Pseudo-Random Function
+    // We assign mode 0x03 for the PPOPRF
+    let context_string =
+      format!("{}-{}-{}", "PPOPRFv1", 0x03, "ristretto255-strobe");
+    //seedDST = "Seed-" || contextString
+    let seed_dst = format!("{}-{}", "Seed", context_string);
+
+    // seedTranscript = I2OSP(len(Bm), 2) || Bm || I2OSP(len(seedDST), 2) || seedDST
+    let mut seed_transcript = Vec::new();
+    seed_transcript.extend_from_slice(&ProofDLEQ::i2osp2(COMPRESSED_POINT_LEN)); //len(Bm)
+    seed_transcript.extend_from_slice(b.compress().as_bytes()); //Bm
+    seed_transcript.extend_from_slice(&ProofDLEQ::i2osp2(seed_dst.len())); //len(seedDST)
+    seed_transcript.extend_from_slice(seed_dst.as_bytes()); //seedDST
+
+    let mut seed = [0u8; DIGEST_LEN];
+    strobe_hash(&seed_transcript, "Seed", &mut seed);
+
+    //M = G.Identity()
+    let mut m = RistrettoPoint::identity();
+    //Z = G.Identity()
+    let mut z = RistrettoPoint::identity();
+
+    let compressed_point_len_slice = &ProofDLEQ::i2osp2(COMPRESSED_POINT_LEN);
+    // for i in range(m):
+    for i in 0..c.len() {
+      //compositeTranscript = I2OSP(len(seed), 2) || seed || I2OSP(i, 2) ||
+      //                      I2OSP(len(C[i]), 2) || C[i] || I2OSP(len(D[i]), 2) || D[i] || "Composite"
+      let mut composite_transcript = Vec::new();
+      composite_transcript.extend_from_slice(&ProofDLEQ::i2osp2(seed.len())); //len(seed)
+      composite_transcript.extend_from_slice(&seed); //seed
+      composite_transcript.extend_from_slice(&ProofDLEQ::i2osp2(i)); //len(i)
+      composite_transcript.extend_from_slice(compressed_point_len_slice); //len(C[i])
+      composite_transcript.extend_from_slice(c[i].compress().as_bytes()); //C[i]
+      composite_transcript.extend_from_slice(compressed_point_len_slice); //len(D[i])
+      composite_transcript.extend_from_slice(d[i].compress().as_bytes()); //D[i]
+
+      //di = G.HashToScalar(compositeTranscript)
+      let di = ProofDLEQ::hash_to_scalar(&composite_transcript, "Composite");
```

Comment:
For hashing to a scalar, we are using everywhere else in the code this: https://github.com/brave/sta-rs/blob/main/ppoprf/src/ppoprf.rs#L59, so I'll follow the same approach.

Reply:
I removed the `hash` function, as it was used nowhere else in the code.